Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -351,14 +386,13 @@ def get_bins(workspace, wkspIndex, withDy=False):
     :param withDy: if True, it will return the error in the "counts", otherwise None
 
     """
-    num_hist = workspace.getNumberHistograms()
-    x = range(0, num_hist)
+    x = get_detectors_indices(workspace)
     y = []
```

Comment:
if y is just the y values then I'm kind of ok with that, but would prefer y_values as it indicates that it is a list.